### ISSUE TITLE
Training loss curves?

### ISSUE THREAD

**Issue by shinshiner** [TencentARC/Open-MAGVIT2#18]
Hello, thanks your great reproduction. However, as we all known, GAN is typically hard to train and may be influenced by the training process, especially when changing some hyper-params, like batch size.

Could you kindly share your loss curves of adversial and reconstruction for reference? 

BTW, why not train generator first and use discriminator later (maybe 10k, 20k) like llamagen does? does it harm to gan training?

**Comment by LouisSerrano** [comment]
I would also be interested 👍

**Comment by RobertLuo1** [comment]
Hi! @shinshiner and @LouisSerrano Thanks for your interest on our work. 
```Could you kindly share your loss curves of adversial and reconstruction for reference?```
We feel sorry that we have no plan to release the loss curve as we have already provided the training configuration. 
```BTW, why not train generator first and use discriminator later (maybe 10k, 20k) like llamagen does? does it harm to gan training?```
In our experiments, we observe that the training of G and D is stable. It indicates that there is no need first train G with some iterations to stabilize the training.

**Comment by LouisSerrano** [comment]
Thanks for your reply. I am actually asking this because I have never trained a GAN or VQGAN before, and I have not found yet loss curves for prior trainings. I am not training the model on image data, and I wanted to know roughly what to expect with the different losses. Also, I noticed that the discriminator had relatively few parameters compared to the generator (about 3 millions), is this usual ? 
Thanks again for your great implementation !

**Comment by RobertLuo1** [comment]
Hi, @LouisSerrano, 
1. I think you can observe the tendency of different training losses when you use our default training configuration. And you can have a rough perception of each loss. 
2. Yep, since the discriminator only needs to judge the fake or real samples, it is not necessary to enlarge it to the comparable parameters as the generator has.

**Comment by LouisSerrano** [comment]
OK thanks for the reply, I'll make sure to reproduce your setting to better understand the training process 👍
